generator/src/main/java/com/scottlogic/deg/generator/decisiontree/ProfileDecisionTreeFactory.java

```diff
@@ -141,9 +141,12 @@ else if (violatedConstraint instanceof ConditionalConstraint) {
             );
         }
 
-        // we've got an atomic constraint
-        AtomicConstraint violatedAtomicConstraint = (AtomicConstraint) violatedConstraint;
-        return convertConstraint(new ViolatedAtomicConstraint(violatedAtomicConstraint.negate()));
+        if (violatedConstraint instanceof AtomicConstraint) {
+            AtomicConstraint violatedAtomicConstraint = (AtomicConstraint) violatedConstraint;
+            return convertConstraint(new ViolatedAtomicConstraint(violatedAtomicConstraint.negate()));
+        }
+
+        return convertConstraint(violatedConstraint);
```

Comment:
if convertConstraint(ViolateConstraint constraintToConvert) 
gets passed in a (violate(not(something))

it will get to this point and then return calling 
convertConstraint(not(something))

which then won't violate

i believe..

Reply:
A NotConstraint is an AtomicConstraint so that shouldn't occur

Reply:
oh, as in, 

you can violate a not constraint

which should get you a violated not constraint.

but in this, the violate(not(constraint)) seems to just get returned as a not(constraint) and fails to be violated